File: 프로그래머스/3/70130. 스타 수열/스타 수열.py
from collections import Counter


def solution(a):
    answer = -1
    counter = Counter(a)
    
    for k in counter:
        if counter[k] <= answer:
            continue
        
        cnt = 0
        i = 0
        
        while i < len(a) - 1:
            if a[i] == a[i + 1] or (a[i] != k and a[i + 1] != k):
                i += 1
            else:
                i += 2
                cnt += 1
    
        answer = max(answer, cnt)
    
    return -1 if answer < 0 else answer * 2

File: 프로그래머스/3/70130. 스타 수열/스타 수열.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    
    public int solution(int[] a) {
        int answer = -1;

        Map<Integer, Integer> counter = new HashMap<>();
        for (int num : a) {
            counter.put(num, counter.getOrDefault(num, 0) + 1);
        }

        for (int k : counter.keySet()) {
            if (counter.get(k) <= answer) continue;

            int cnt = 0;
            int i = 0;

            while (i < a.length - 1) {
                if (a[i] == a[i + 1] || (a[i] != k && a[i + 1] != k)) {
                    i += 1;
                } else {
                    i += 2;
                    cnt += 1;
                }
            }

            answer = Math.max(answer, cnt);
        }

        return answer < 0 ? -1 : answer * 2;
    }
}